openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -533,3 +535,33 @@ def _validate_email(email):
         raise AccountEmailInvalid(
             u"Email '{email}' format is not valid".format(email=email)
         )
+
+
+@intercept_errors(UserAPIInternalError, ignore_errors=[UserAPIRequestError])
+@transaction.atomic
+def delete_user(user):
+    """
+    Delete a user and any related records.
+
+    Related records include the following:
+    1. models with a ForeignKey relationship to User (with on_delete=CASCADE)
+    2. models with indirect relationship to User (i.e. through another model, or with on_delete other than CASCADE)
+    3. files belonging to the user
+
+    Arguments:
+        user: The User object
+
+    Returns:
+        None
+
+    Raises:
+        UserAPIInternalError
+    """
+    # delete profile images
+    delete_profile_images.delay([user.username])
+
+    # retire user discussion comments
```

Comment:
There's a small overlap here with [this code](https://github.com/edx-solutions/api-integration/blob/master/edx_solutions_api_integration/users/views.py#L388) which already retires discussion users and delete all models.

I'm not sure why all these were added here.

Also, I don't see this being used anywhere, do you know if there's an API endpoint for this?

Reply:
@viadanna, [this PR](https://github.com/edx-solutions/api-integration/pull/197) mentioned in the related PRs section of the description has the changes to `api-integration` to use this. John Louis had also updated the existing code there to move the deletion of the discussion comments to this method. Is that okay or should I retain the current behaviour of doing the deletion there?

Reply:
I think it would make sense to move it here for upstreaming purposes.

My only concern is that [this query](https://github.com/edx-solutions/api-integration/pull/197/files#diff-6e10f5856db69f12fe0c2aef7e4ec7dfL403) should be more efficient than performing one delete query for each user when removing a company or performing bulk-deletion.

Do you think the behavior implemented in api-integration can be replicated here?

Reply:
@viadanna, I have updated this PR and the related `api-integration` PR to move all the deletion logic to the `delete_users` method in this PR. It also deletes all the users using the queryset instead of deleting the users one by one.